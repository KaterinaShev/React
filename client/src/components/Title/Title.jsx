import React from 'react'
import PropTypes from 'prop-types'
import styles from './Title.module.scss'

const Title = ({ title, subtitle, showContent, className }) => {
  return (
    <>
      {title && (
        <h5 onClick={showContent} className={styles.title + ' ' + className}>
          {title}
        </h5>
      )}
      {subtitle && <h1 className={styles.subtitle}>{subtitle}</h1>}
    </>
  )
}

Title.propTypes = {
  title: PropTypes.string,
  subtitle: PropTypes.string
}

export default Title
