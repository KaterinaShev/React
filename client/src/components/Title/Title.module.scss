@import '../../styles/variables.scss';
@import '../../styles/breakpoints.scss';

.filtering {
  display: none;
  @include for-medium-up {
    display: block;
    cursor: pointer;
  }
}
.subtitle {
  font-weight: 400;
  font-size: 53px;
  line-height: 82px;
  margin: 35px 0 45px;
  @include for-large-up {
    font-size: 48px;
    line-height: 61px;
    margin: 30px 0 25px;
  }
  @include for-small-up {
    font-size: 28px;
    margin: 15px 0;
  }
}
.title {
  color: #97a9ba;
  font-size: 16px;
  line-height: 22px;
  text-transform: uppercase;
  margin-top: 100px;
  @include for-small-up {
    margin-top: 50px;
  }
}
