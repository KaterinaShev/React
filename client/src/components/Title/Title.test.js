import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import Title from './Title'

const showContent = jest.fn()

describe('Title', () => {
  test('Title is render', () => {
    const { asFragment } = render(<Title showContent={showContent}/>)
    expect(asFragment()).toMatchSnapshot()
  })
  test('Title with props title', () => {
    const { getByText } = render(<Title title='Something' />)
    const linkOnElement = getByText(/Something/i)
    expect(linkOnElement).toBeInTheDocument()
  })
  test('Title with props subtitle', () => {
    const { getByText } = render(<Title subtitle='Something more' />)
    const linkOnElement = getByText(/Something more/i)
    expect(linkOnElement).toBeInTheDocument()
  })
})
