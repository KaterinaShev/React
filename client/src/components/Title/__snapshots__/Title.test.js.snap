// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Title is render 1`] = `<DocumentFragment />`;

exports[`Title Title with showContent 1`] = `<DocumentFragment />`;
